### ISSUE TITLE
Authelia with Traefik and Setup Question

### ISSUE THREAD

**Issue by hungry-bogart** [tamimology/cloudflare-authelia#2]
I have two quick questions.

1. Most compose files to set up Authelia include Traefik. Cloudflare has its own reverse proxy system, so is Traefik even needed?
2. I assume that it's important to have Authelia setup prior to following the guide. Is that correct?

**Comment by tamimology** [comment]
2-Yes it is always safer to have all your exposed network/containers behind a 2fa platform. I choose Authelia as I found it reliable and does the job with CF and NGINX if needed

1-Traefik is a reverse proxy where you need to expose ports by using port forwarding, similar to NPM. It is not recommended to port forward is you are opening the most vulnerable ports (443 and 80). CF does not require any port forwarding, as it has its built-in tunnel, called the argo tunnel. So you just redirect your SUBDOMAIN.DOMAIN within CF to your LOCALIP:PORT from the CF platform, with nothing to be done in the router. When doing so, you need to include 2FA as well, this guide shows you how to integrate Authelia with CF without using any reverse proxies.


Hope those explain enough and answers your questions. By the way, I used NPM + Authelia + port forwarding. Then realised all the above and moved everything to CF+Authelia, hence I wrote this guide as it took me some time to integrate Authelia as there is no clear guide on the internet.

**Comment by hungry-bogart** [comment]
Thanks for putting this guide together. I was looking for something exactly like this and I'm right in the middle of getting it set up. I appreciate your explanation here. I've used NPM and still do on a VPS deployment, but I wanted to try something different and that's how I ended up with Cloudflare.

I have also tried Tailscale and Twingate for a few similar projects. It's good to see that there are solid options available.
